detox/index.d.ts

```diff
@@ -145,16 +158,28 @@ declare global {
             enabled?: boolean;
         }
 
+        interface DetoxIosAppConfig extends DetoxLooseIosAppConfig {
+            type: 'ios.app';
+        }
+
         interface DetoxLooseIosAppConfig {
             binaryPath: string;
+            bundleId?: string;
             build?: string;
+            launchArgs?: Record<string, string | null | undefined>;
+        }
+
+        interface DetoxAndroidAppConfig extends DetoxLooseIosAppConfig {
+            type: 'android.apk';
         }
 
         interface DetoxLooseAndroidAppConfig {
```

Comment:
I understand where the loose concept is coming from but since they are never directly referenced - only the concrete types do, then maybe it'd be best to simplify by unifying everything onto the concrete types and that's it?

Or alternatively, create one "loose" config that includes 'em properties that are relevant to both (i.e. all) platforms (e.g `launchArgs`)? And then reconsider the name loose, and try to go with conventions such as "abstract" or just "base"?... 🤔

Reply:
Mmm, I can see your point - let me check one more time.